### PR TITLE
Remove Amazon deprecated modules

### PR DESCRIPTION
**Pull request by vincbeck**
A new major release of the Amazon provider package will happen soon. Let's take advantage of it to remove all Amazon deprecated modules

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/aws_lambda.py [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-#
-# Unless required by applicable law or agreed to in writing,
-# software distributed under the License is distributed on an
-# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-# KIND, either express or implied.  See the License for the
-# specific language governing permissions and limitations
-# under the License.
-"""This module is deprecated. Please use :mod:`airflow.providers.amazon.aws.operators.lambda_function`."""
-
-import warnings
```
Comment (eladkal) [inline airflow/providers/amazon/aws/operators/aws_lambda.py]: This one was just deprecated in https://github.com/apache/airflow/pull/25100 and was not released yet.
Are you sure you want to remove it now?
This means that for this module users will have no early warning.

Comment (vincbeck) [reply airflow/providers/amazon/aws/operators/aws_lambda.py]: That's valid point, let's keep it

**Review by potiuk — APPROVED** [review]
Pending test success.

**Comment by josh-fell** [comment]
@vincbeck Curious if other modules that were deprecated because they were renamed like [`EKSPodOperator`](https://github.com/apache/airflow/blob/dfc943054abfdf2a182ee6345498c2487ed3c0f7/airflow/providers/amazon/aws/operators/eks.py#L816) be included as well in the major release deprecation?

**Comment by vincbeck** [comment]
You're right! It seems I missed a lof of deprecated modules ... Sorry about that. I'll try to come up with a fast PR to include it in the release as well

**Comment by vincbeck** [comment]
#25609

**Pull request merged**